lib/Elastica/Script/AbstractScript.php

```diff
@@ -2,28 +2,160 @@
 namespace Elastica\Script;
 
 use Elastica\AbstractUpdateAction;
+use Elastica\Exception\InvalidException;
 
 /**
  * Base class for Script object.
  *
+ * Wherever scripting is supported in the Elasticsearch API, scripts can be referenced as "inline", "id" or "file".
+ *
  * @author Nicolas Assing <nicolas.assing@gmail.com>
+ * @author Tobias Schultze <http://tobion.de>
+ * @author Martin Janser <martin.janser@liip.ch>
  *
  * @link https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-scripting.html
  */
 abstract class AbstractScript extends AbstractUpdateAction
 {
+    const LANG_MVEL = 'mvel';
+    const LANG_JS = 'js';
+    const LANG_GROOVY = 'groovy';
+    const LANG_PYTHON = 'python';
+    const LANG_NATIVE = 'native';
+    const LANG_EXPRESSION = 'expression';
+    const LANG_PAINLESS = 'painless';
+
     /**
-     * @param array|null $params
-     * @param string     $id
+     * @var string
      */
-    public function __construct(array $params = null, $id = null)
```

Comment:
I also assume that everyone that used it hopefully contributed it back to Elastica. For completeness I would still like to mention it under breaking changes. That doesn't mean I will not create 5.1 out of it ;-)